frontend/src/components/ProjectSelector.tsx

```
@@ -0,0 +1,44 @@
+    onRefresh: () => void;
+}
+
+export function ProjectSelector({ currentProjectUrl, onSelect, projects, loading, error, onRefresh }: ProjectSelectorProps) {
+    if (loading) {
+        return <div className="text-xs text-gray-500">Loading projects...</div>;
+    }
+    if (error) {
+        return (
+            <div className="text-xs text-red-500 flex items-center space-x-2">
+                <span>{error}</span>
+                <button className="underline text-blue-600" onClick={onRefresh}>Retry</button>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)